### ISSUE TITLE
Please do not bundle cort-test.el

### ISSUE THREAD

**Issue by conao3** [conao3/seml-mode.el#111]
Original issue is [here](https://gitlab.com/oer/org-re-reveal/issues/15) from @tarsius

> @conao3 Please do not add your testing utilities cort-test.el to multiple repositories. Such copies tend to go out of sink and it is undefined which version gets loaded if there are more than one on the load-path.
Instead please release this as a separate package, which probably should be maintained in its own repository.

**Comment by conao3** [comment]
I developed `cort-test.el` to ensure that some package I developed worked from Emacs-22, and for better debug messages and concise test definitions than `ert.el`.
It is because of the policy of `cort-test.el` that it is independent of one file and appears to be scattered across many repositories.

`cort-test.el` is used only for testing, purely by `emacs --batch`, and does not need to be loaded into Emacs, which is always running (You are operating it.).
I'm loading it, but it's just loading because it's inconvenient for setting indents and jumping definitions.

Each `cort-test.el` can fulfill its own responsibilities for each version, and bundling `cort-test.el` into each repository is essential so that if the test definitions that `cort-test.el` accepts change, it does not corrupt the CIs of that package.

I think your suggestion is the will of MELPA, but MELPA ignores `*-test.el` files by default and does not bother many Emacs users.
In other words, it's not a test package, it's just a script file for defining and running tests.

My intention is that it is necessary to resolve the package dependencies for unit testing, but I don't like downloading obscure, cluttered files for testing. It will no longer be called a unit test.

cf: cort-test.el managed [here](https://github.com/conao3/cort-test.el)

**Comment by conao3** [comment]
For example, the `cort-test.el` bundled with this package is v 4.0.0 and does not satisfy upstream's request to update the test file and `Makefile`.
If you download `cort-test.el` from upstream, the CI will be broken, and the developer will be confused and suspect that each package is broken.
In fact, the test definition expected by the `cort-test.el` was just changed.

https://github.com/conao3/cort-test.el#v40-to-v50
https://github.com/conao3/cort-test.el#v50-to-v60

**Comment by tarsius** [comment]
I am aware that Melpa excludes test files. This request doesn't really concern Melpa, but the [Emacsmirror](https://emacsmirror.net) and users who install packages by cloning repositories using e.g. my borg.el or straight.el.

As part of my work on the Emacsmirror I try to ensure that all dependencies of all mirrored packages are satisfied and that no two packages provide the same feature. Considering how many packages are mirrored the number of remaining issues and special cases is fairly small, as can be seen [here](https://emacsmirror.net/stats/kludges.html).

Given your explanation I think you should continue to bundle this file but stop providing a feature. Then instead of using `require` use `load` to load the library. That way you can guarantee that the correct version is loaded when *you* run the tests. I recommend that no test files provide a feature, even if there is no risk of conflict with another "library" from another package because the package prefix is being used.

**Comment by conao3** [comment]
> but stop providing a feature

I agree with you on this point. I will send PR to all repositories that bundle this file (just some my packages and `org-re-reveal`) and correct this.

**Comment by tarsius** [comment]
Thanks a lot!

**Comment by conao3** [comment]
Thanks for your suggestion!
